app/javascript/packages/address-search/hooks/use-validated-usps-locations.ts

```diff
@@ -14,6 +15,7 @@ export default function useValidatedUspsLocations(locationsURL: string) {
   const checkValidityAndDisplayErrors = (address, city, state, zipCode) => {
     let formIsValid = true;
     const zipCodeIsValid = zipCode.length === 5 && !!zipCode.match(/\d{5}/);
+    const addressIsValid = address.replace(/[A-Za-z0-9\-' .#]*/g, '') === '';
```

Comment:
Looks like both files would have access to validatedAddressFieldRef since this one creates it, and the other one just uses it. Can we arbitrarily set the pattern as a property on the ref in this file? Or is there a specific way of handling that? Not super familiar with ref uses.

Reply:
I was expecting that we could keep the `pattern` assignment in the JSX for the `<TextInput />` element, and then retrieve it off the ref here. `validatedAddressFieldRef.current` should be the rendered [`HTMLInputElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement)..

Reply:
Oh! I was thinking about it the wrong way around. Great idea!